### DISCUSSION TITLE
Use quickadd macro to get all files within folder (project management)

### DISCUSSION THREAD

**Discussion by imed-ghomari** [chhoumann/quickadd#113] [Q&A]
### What I'm trying to do
I want to use this plugin for project management. Each project have its own file under the project folder. Now, whenever I have a new idea :
- it is appended to a related project/file
- it is added to a new note if it doesn't have a related project

![image](https://user-images.githubusercontent.com/38974541/134973032-8e138f0b-3d42-49c4-9454-891d1f8a7283.png)

### Things I have tried
I have tried to use "capture", and I can successfully create a new note or append my idea to an already existing file, but here is the problem :
- **Problem :** I have to manually write each file name as a proposition like this `project folder/{{VALUE:file1,file2,file3...}}.md` 
- **Proposed solution :** I was thinking something like `project folder/{{VALUE:{{macro:macroname}}}}.md` and the macro would create a comma separated list of all the files under my project folder

![image](https://user-images.githubusercontent.com/38974541/134973056-e8598dad-61c9-4362-9363-4191203a645b.png)

Can this be achieved ? (alternative solutions are welcome)

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

What you described is a built-in feature now, no macro hack needed. Set the Capture's _Capture To_ to your project folder:

```
project folder/
```

QuickAdd then opens a suggester listing every markdown file in that folder (nested folders included), so appending an idea to the right project is just picking it from the list. For the "no related project yet" branch: enable _Create file if it doesn't exist_, and typing a new name in the suggester creates that note in the folder.

Docs: https://quickadd.obsidian.guide/docs/Choices/CaptureChoice#capturing-to-folders

Closing as resolved.
